### PR TITLE
chore: improve compatibility of EXPIRE functions with Redis

### PR DESCRIPTION
**Pull request by romange**
Also, provide a module name if stumbled upon module data that can not be loaded by dragonfly.

### CODE REVIEW

File: src/server/rdb_load.cc [hunk 1]
```diff
@@ -203,6 +203,25 @@ int ziplistPairsConvertAndValidateIntegrity(const uint8_t* zl, size_t size, unsi
   return ret;
 }
 
+static string ModuleTypeName(uint64_t module_id) {
```
Comment (kostasrim) [inline src/server/rdb_load.cc:206]: I think this is static anyway as it's part of an unamed namespace

File: src/server/generic_family.cc [hunk 2]
```diff
@@ -851,7 +853,13 @@ void GenericFamily::Pexpire(CmdArgList args, ConnectionContext* cntx) {
   if (!absl::SimpleAtoi(msec, &int_arg)) {
     return cntx->SendError(kInvalidIntErr);
   }
-  int_arg = std::max<int64_t>(int_arg, 0L);
+  int_arg = std::max<int64_t>(int_arg, -1);
+
+  // to be more compatible with redis, we silently cap the expire time to kMaxExpireDeadlineSec
+  if (int_arg > kMaxExpireDeadlineSec * 1000) {
```
Comment (adiholden) [inline src/server/generic_family.cc:859]: Is there a reason to do this only in this fuctions? as we set expire in other commands such as SetEx

File: src/server/rdb_load.cc [hunk 3]
```diff
@@ -203,6 +203,25 @@ int ziplistPairsConvertAndValidateIntegrity(const uint8_t* zl, size_t size, unsi
...
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz"
+      "0123456789-_";
+
+  char name[10];
+
+  name[9] = '\0';
+  char* p = name + 8;
+  module_id >>= 10;
+  for (int j = 0; j < 9; j++) {
```
Comment (adiholden) [inline src/server/rdb_load.cc:217]: how did you check this? do you have rdb with modules?

Comment (romange) [reply src/server/rdb_load.cc:217]: see here: https://github.com/redis/redis/blob/6.2/src/module.c#L4386

Comment (romange) [reply src/server/rdb_load.cc:217]: I created manually with redis-stack. see attached.

[dump.rdb.gz](https://github.com/dragonflydb/dragonfly/files/14529277/dump.rdb.gz)

Comment (romange) [reply src/server/rdb_load.cc:217]: it does not contain a module. it's module data, for example json entries are stored under module name `scdtype00`

**Review by adiholden — APPROVED** [review]

**Pull request merged**